geowebcache/core/src/main/java/org/geowebcache/GeoWebCacheTypedEnvironment.java

```diff
@@ -0,0 +1,69 @@
+/**
+ * This program is free software: you can redistribute it and/or modify it under the terms of the
+ * GNU Lesser General Public License as published by the Free Software Foundation, either version 3
+ * of the License, or (at your option) any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * <p>You should have received a copy of the GNU Lesser General Public License along with this
+ * program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * @author Fernando Mino, GeoSolutions S.A.S., Copyright 2019
+ */
+package org.geowebcache;
+
+import java.util.Optional;
+import org.apache.commons.lang3.StringUtils;
+
+/** Enviroment variables placeholder resolver with declared types. */
+public class GeoWebCacheTypedEnvironment {
```

Comment:
I don't see a need for a new class, an extra static method in GeoWebCacheEnvironment should do the trick.

Reply:
Indeed, merged new method on GeoWebCacheEnvironment class.